test/unit/pages/TideDashboard.test.js

```diff
@@ -692,6 +731,152 @@ describe('TideDashboard', () => {
+              { type: 'SET_CLINIC_PATIENT_LAST_REVIEWED_REQUEST' },
+              {
+                type: 'SET_CLINIC_PATIENT_LAST_REVIEWED_SUCCESS',
+                payload: { clinicId: 'clinicID123', patientId: 'ea8ab4da-d4ed-bc6a-dec3-6aa170a99d49', reviews: [today, yesterday] },
+              },
+            ]);
+
+            done();
+          });
+        });
+
+        it('should allow undoing last reviewed date', done => {
+          const table = wrapper.find('table#dashboard-table-meetingTargets');
+          const rows = table.find('tbody tr');
+          const lastReviewData = row => rows.at(row).find('.MuiTableCell-root').at(10);
+          const updateButton = () =>lastReviewData(4).find('button');
+
+          expect(lastReviewData(4).text()).to.contain('Today');
+          expect(updateButton().text()).to.equal('Undo');
+
+          store.clearActions();
+          updateButton().simulate('click');
+          setTimeout(() => {
+            sinon.assert.calledWith(defaultProps.trackMetric, 'Clinic - Undo mark patient reviewed', sinon.match({ clinicId: 'clinicID123', source: 'TIDE dashboard' }));
+
+            sinon.assert.calledWith(
+              defaultProps.api.clinics.revertClinicPatientLastReviewed,
+              'clinicID123',
+            );
+
+            expect(store.getActions()).to.eql([
+              { type: 'REVERT_CLINIC_PATIENT_LAST_REVIEWED_REQUEST' },
+              {
+                type: 'REVERT_CLINIC_PATIENT_LAST_REVIEWED_SUCCESS',
+                payload: { clinicId: 'clinicID123', patientId: 'ecabff50-0698-ec3d-f2b9-b9d3720cbe14', reviews: [yesterday] },
+              },
+            ]);
+
+            done();
+          });
+        });
+
+        it('should not render the Last Reviewed column if showSummarData flag is false', () => {
+          TideDashboard.__Rewire__('useFlags', sinon.stub().returns({
+            showSummaryData: false,
+          }));
+
+          store = mockStore(hasResultsState);
+          wrapper = mount(
+            <Provider store={store}>
+              <ToastProvider>
+                <TideDashboard {...defaultProps} />
+              </ToastProvider>
+            </Provider>
+          );
+
+          const lastReviewedHeader = wrapper.find('#dashboard-table-meetingTargets-header-lastReviewed').hostNodes();
+          expect(lastReviewedHeader).to.have.length(0);
+
+          TideDashboard.__ResetDependency__('useFlags');
+        });
+      });
+
+      context('showSummaryDashboardLastReviewed flag is false', () => {
+        beforeEach(() => {
+          TideDashboard.__Rewire__('useFlags', sinon.stub().returns({
+            showSummaryDashboardLastReviewed: false,
+          }));
+        });
+
+        afterEach(() => {
+          TideDashboard.__ResetDependency__('useFlags');
+        });
+
+        it('should not show the Last Reviewed column, even if clinic tier >= tier0300', () => {
+          store = mockStore(hasResultsState);
```

Comment:
Should this be the `tier0300ClinicState`?

Reply:
Yep.  Will fix